### PR TITLE
Pokemon stats

### PR DESCRIPTION
**Pull request by yung-coder**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

So here I have tried to build a pokemon stats app that will give you information about a particular pokemon on search.

Fixes # (issue)
#527 

## Type of change

- [ ✔️] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

it has been tested by me in my local system for all the cases.

# Checklist:

- [✔️ ] I have performed a self-review of my own code
- [✔️ ] I have commented my code, particularly in hard-to-understand areas
- [✔️ ] I have made corresponding changes to the documentation
- [✔️ ] My changes generate no new warnings
- [✔️ ] I have added tests that prove my fix is effective or that my feature works
- [✔️ ] New and existing unit tests pass locally with my changes
- [✔️ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/plays/pokemon-stats/PokemonStats.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import "./styles.css";
+import axios from "axios";
+import Pokemoncard from "./components/Pokemoncard";
+import Search from "./components/Search";
+// WARNING: Do not change the entry componenet name
+function PokemonStats(props) {
+  // Your Code Start below.
+  const [pokemon, setpokemon] = useState();
+  const [name, setname] = useState("");
+  const Pokemon = (e) => {
```
Comment (atapas) [inline src/plays/pokemon-stats/PokemonStats.js:12]: This method name can be like fetchPokemonDetails. The convention used look like we are creating a class.

File: src/plays/pokemon-stats/components/Pokemoncard.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      <div class="wrapper">
+        <div class="pokemon-card pokemon">
+          <div class="pokemon-card__image pokemon-card__image--pokemon">
+            <img src={pokemon.sprites.front_default} alt="pokemon" />
+          </div>
+          <div class="poke-name">{pokemon.species.name}</div>
+
+          <div class="pokemon-stats">
+            <div class="one-third">
+              <div class="stat">HP:</div>
```
Comment (atapas) [inline src/plays/pokemon-stats/components/Pokemoncard.jsx:16]: May be good to expand some of these terms.. for the non pokemon lovers :)

**Review by atapas — CHANGES_REQUESTED** [review]
A few initial comments

**Comment by atapas** [comment]
@yung-coder 

is it under development? I have given a few comments. Apart from it, I am getting some errors in the console too..The search doesn't work for me

image

**Comment by yung-coder** [comment]
Write the name of pokemon in small letters it will work...

@atapas

**Comment by atapas** [comment]
> Visit Preview

I see :)

Then you fix that to lower the name always before passing to the API.. Let user type in case sensitive way.

Also, a couple of things,

- You may want to handle the error case when something doesn't;t have info or invalid
- The Pokemon description card... the info looks as clickable.

**Comment by yung-coder** [comment]
Okay i will handle it as case sensitive and for card description the info is clickable for details  should i do some specific changes....?

**Comment by atapas** [comment]
> Okay i will handle it as case sensitive and for card description the info is clickable for details should i do some specific changes....?

Are they really meant to be clickable? Because when I click on them, nothing happens. So if they are not supposed to be clickable, change the cursor stye to none.

**Comment by yung-coder** [comment]
Yeah nothing will happen it will just change the color on hover ...

**Comment by atapas** [comment]
> Yeah nothing will happen it will just change the color on hover ...

OK

**Comment by yung-coder** [comment]
So, should i remove cursor pointer or not...

**Comment by atapas** [comment]
> So, should i remove cursor pointer or not...

Yes Yes Yes :)

**Comment by yung-coder** [comment]
Hey, @atapas I have made the changes pls give it a look .....

**Review by koustov — CHANGES_REQUESTED** [review]
I haven't jumped into code however here are few observations.
1. If do not type anything and hit search, I get 
image
2. I am not a pokemon follower so I really don't know any name other than Pikachu. So cant type. Shall we have some drop-down or type ahead for exploring?
3. When I type, upon hitting enter should search the text. I mean, that should be a `submit` button.
4. Once a stat is shown, how do I search for another one?
5. In the mobile resolution, I am feeling like losing padding (iPhone SE)

**Comment by yung-coder** [comment]
For button type sumbit i will do it and for the feature of drop-down surely it's a good idea but i don't have any idea how can we implement it so if you have an idea please let me know and for resolutions i will look at it and if we once search a pokemon name we will be shown it's stats like hp and etc....if we want to search for another we have to go back and then we can search......


@koustov

**Comment by yung-coder** [comment]
Hey, @atapas and @koustov please give it a look I have made the changes I think it's good but let me know if you want any modifications ...

File: package.json [hunk 3]
```diff
@@ -41,6 +41,7 @@
     "react-scripts": "5.0.0",
     "react-shimmer-effect": "^1.0.9",
     "react-simple-maps": "^2.3.0",
+    "react-toastify": "^9.0.8",
```
Comment (atapas) [inline package.json:44]: Isn't there a toast message lib in already? Please check before adding.

File: src/plays/pokemon-stats/PokemonStats.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  const [name, setname] = useState("");
+  const fetchPokemonDetails = (e) => {
+    e.preventDefault();
+    axios
+      .get(`https://pokeapi.co/api/v2/pokemon/${name.toLowerCase()}`)
+      .then((response) => {
+        setpokemon(response.data);
+      })
+      .catch((err) => {
+        toast.error("Sorry, Unable to fetch", {
```
Comment (atapas) [inline src/plays/pokemon-stats/PokemonStats.js:22]: Give a better comment like, the Pokemon you are searching is not found, try another.

**Review by atapas — CHANGES_REQUESTED** [review]

File: src/plays/pokemon-stats/PokemonStats.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+          draggable: true,
+          progress: undefined,
+        });
+      });
+  };
+  return (
+    <>
+      <div className="play-details">
+        <PlayHeader play={props} />
+        <div className="Skeleton-pokemon-app">
```
Comment (koustov) [inline src/plays/pokemon-stats/PokemonStats.js:37]: `Skeleton-pokemon-app` should be in small letters

Comment (koustov) [inline src/plays/pokemon-stats/PokemonStats.js:37]: You should add this line after `{/* Your Code Starts Here */}`

File: src/plays/pokemon-stats/Readme.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+- Level: Intermediate
+
+## Creator Information
+
+- User: yung-coder
+- Gihub Link: https://github.com/yung-coder
+- Blog: 
+- Video: 
+
+## Implementation Details
```
Comment (koustov) [inline src/plays/pokemon-stats/Readme.md:17]: Give some implementation details

File: src/plays/pokemon-stats/Readme.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+Update your implementation idea and details here
+
+## Consideration
```
Comment (koustov) [inline src/plays/pokemon-stats/Readme.md:21]: remove the section if not applicable

File: src/plays/pokemon-stats/Readme.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+Update all considerations(if any)
+
+## Resources
```
Comment (koustov) [inline src/plays/pokemon-stats/Readme.md:25]: remove the section if not applicable

File: src/plays/pokemon-stats/components/pokecard.css [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+.pokemon-card__level {
+  text-transform: uppercase;
+  font-size: 12px;
+  font-weight: 700;
+  margin-bottom: 3px;
+}
+
+.pokemon-card__level--pokemon {
```
Comment (koustov) [inline src/plays/pokemon-stats/components/pokecard.css:53]: Do you wish to use scss. It is a lot easier to maintain nesting classes.

File: src/plays/pokemon-stats/components/Search.jsx [hunk 10]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import "./search.css";
+const search = (props) => {
+  const { setname, fetchPokemonDetails, name } = props;
+  return (
+    <div className="poke-main-cont">
+      <div className="poke-input-cont field">
+        <form onSubmit={fetchPokemonDetails} className="poke-form">
```
Comment (koustov) [inline src/plays/pokemon-stats/components/Search.jsx:8]: The still does not honor `Enter` button. I have to explicitly click on it.

File: src/plays/pokemon-stats/PokemonStats.js [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+          {/* Your Code Starts Here */}
+          <ToastContainer
```
Comment (koustov) [inline src/plays/pokemon-stats/PokemonStats.js:39]: Some random text search didn't show me the toast however the console is recording the problem
image

File: src/plays/pokemon-stats/components/search.css [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+    rgba(45, 35, 66, 0.3) 0 7px 13px -3px, #d6d6e7 0 -3px 0 inset;
+  box-sizing: border-box;
+  color: #36395a;
+  cursor: pointer;
+  display: inline-flex;
+  font-family: "JetBrains Mono", monospace;
+  height: 35px;
+  justify-content: center;
+  line-height: 1;
+  list-style: none;
```
Comment (koustov) [inline src/plays/pokemon-stats/components/search.css:84]: What is the use of `list-style` here?

File: src/plays/pokemon-stats/components/search.css [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+  overflow: hidden;
+  padding-left: 16px;
+  padding-right: 16px;
```
Comment (koustov) [inline src/plays/pokemon-stats/components/search.css:87]: in short `padding: 0px 16px`

File: src/plays/pokemon-stats/components/search.css [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+  position: relative;
+  text-align: left;
```
Comment (koustov) [inline src/plays/pokemon-stats/components/search.css:89]: I believe you are looking for center-aligned text. isn't it?

**Review by koustov — CHANGES_REQUESTED** [review]
Few comments

File: package.json (hunk 3 above)
Comment (koustov) [reply package.json:44]: Yes, one more package already exists

**Comment by koustov** [comment]
> For button type sumbit i will do it and for the feature of drop-down surely it's a good idea but i don't have any idea how can we implement it so if you have an idea please let me know and for resolutions i will look at it and if we once search a pokemon name we will be shown it's stats like hp and etc....if we want to search for another we have to go back and then we can search......
> 
> @koustov

- Drop down: Even if you do not get any API to list all pokemon, you can hard code a small list. You can use MUI auto-complete. 
- Search again: I feel the search should be independent from the result. I mean, we can search continuously and get the result instead of refreshing the page or go back and forth.

**Comment by yung-coder** [comment]
Mui auto complete i was not knowing about it i will give it a look and try to implement...


And for the refreshing yeah we can  have search and result together but i thought this card design would look cool if you want to change result display section please let me know...


@koustov

**Comment by koustov** [comment]
> Mui auto complete i was not knowing about it i will give it a look and try to implement...
> 
> And for the refreshing yeah we can have search and result together but i thought this card design would look cool if you want to change result display section please let me know...
> 
> @koustov

And you are right, the card design is definitely cool and I liked the glassy effect. However, having search and the card separately wouldn't make any impact on the glamour, and that's what I feel

**Comment by yung-coder** [comment]
Changes done @koustov @atapas

**Review by koustov — APPROVED** [review]
lgtm

**Comment by yung-coder** [comment]
Hey @atapas is it all good ...?

**Review by atapas — APPROVED** [review]

**Comment by atapas** [comment]
@yung-coder Some conversations are opened. Could you please respond to them and close?

**Comment by yung-coder** [comment]
Done ✔️ @atapas

**Pull request merged**